mathics/autoload/formats/SVG/Export.m

```diff
@@ -4,13 +4,15 @@
 
 
 SVGExport[filename_, expr_, opts___] := 
-  Module[{strm, data}, 
+  Module[{strm, data, p, expr2}, 
     strm = OpenWrite[filename];
     If[strm === $Failed, Return[$Failed]];
-    If[System`$UseSansSerif,
-	    data = StringTake[ToString[MathMLForm[expr]],{23,-8}],
-	    data = StringTake[ToString[MathMLForm[expr]],{23,-8}]];
-    WriteString[strm, "<svg>" <> data <> "</svg>"];
+    expr2 = If[Head[expr]=!=System`Graphics, System`Graphics[{System`Inset[ToString[expr]]}], expr];
+	 expr2= MathMLForm[expr2];
+    data=ToString[expr2];
+    p = StringPosition[data, "data:image/svg+xml;base64"][[1]][[2]];
+    data = StringTake[data  ,{p+2,-19}];
```

Comment:
Magic numbers like `p+2`, `-19` worry me and are likely to be fragile.

Or expressions like `[[1]][[2]]`. What do these mean and do?

Reply:
When you ask for the MathMLForm, the kernel returns you a mglyph, with the svg that you want encoded in b64.
To get the svg, we need to get the string with the svg encoded, which lies to bytes after the place where `data:image/svg+xml;base64` is found. Also, you need to remove the tail of `"><...>` to get something that you can decode.